sparse_strips/vello_hybrid/shaders/sparse_strip_renderer.wgsl

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// A WGSL shader for rendering sparse strips with alpha blending.
+//
+// Each strip instance represents a horizontal slice of the rendered output and consists of:
+// 1. A variable-width region of alpha values for semi-transparent rendering
+// 2. A solid color region for fully opaque areas
+//
+// The alpha values are stored in a texture and sampled during fragment shading.
+// This approach optimizes memory usage by only storing alpha data where needed.
+
+struct Config {
+    width: u32,
+    height: u32,
+    strip_height: u32,
+    alpha_texture_width: u32,
+}
+
+struct StripInstance {
+    // [x, y] packed as u16's
+    @location(0) xy: u32,
+    // [width, dense_width] packed as u16's
+    @location(1) widths: u32,
+    @location(2) col: u32,
+    @location(3) rgba: u32,
+}
+
+struct VertexOutput {
+    @location(0) tex_coord: vec2<f32>,
+    @location(1) @interpolate(flat) dense_end: u32,
+    @location(2) @interpolate(flat) color: u32,
+    @builtin(position) position: vec4<f32>,
+};
+
+@group(0) @binding(1)
+var<uniform> config: Config;
```

Comment:
To be explored separately, I wonder whether it's worth separating `config` out from `alphas_texture` to reduce state change in the pipeline.

Reply:
I think WebGPU can compartmentalise state change if we move less frequently changing resources into lower indexed groups. During a resize operation (i.e. user increases Canvas size each frame), we would need to rebound both the texture and config. If we moved the config to its own group, WebGPU wouldn't need to rebuild its state for the texture too (assuming the config is in group 1 and texture in group 0).

That said, I think it's not worth implementing right now. I think it makes sense as a follow up awhile from now!